include/openssl/core_names.h

```diff
@@ -167,6 +167,28 @@ extern "C" {
 #define OSSL_KDF_NAME_KBKDF         "KBKDF"
 #define OSSL_KDF_NAME_KRB5KDF       "KRB5KDF"
 
+/* Know RAND names */
+#define OSSL_RAND_PARAM_STATUS                  "status"
+#define OSSL_RAND_PARAM_STRENGTH                "strength"
+#define OSSL_RAND_PARAM_RESEED_REQUESTS         "reseed_requests"
+#define OSSL_RAND_PARAM_RESEED_TIME_INTERVAL    "reseed_time_interval"
+#define OSSL_RAND_PARAM_MAX_REQUEST             "max_request"
+#define OSSL_RAND_PARAM_MIN_ENTROPYLEN          "min_entropylen"
+#define OSSL_RAND_PARAM_MAX_ENTROPYLEN          "max_entropylen"
+#define OSSL_RAND_PARAM_MIN_NONCELEN            "min_noncelen"
+#define OSSL_RAND_PARAM_MAX_NONCELEN            "max_noncelen"
+#define OSSL_RAND_PARAM_MAX_PERSLEN             "max_perslen"
+#define OSSL_RAND_PARAM_MAX_ADINLEN             "max_adinlen"
+#define OSSL_RAND_PARAM_RESEED_CTR              "reseed_counter"
+#define OSSL_RAND_PARAM_RESEED_PROP_CTR         "reseed_prop_counter"
+#define OSSL_RAND_PARAM_PROPERTIES              OSSL_ALG_PARAM_PROPERTIES
+#define OSSL_RAND_PARAM_DIGEST                  OSSL_ALG_PARAM_DIGEST
+#define OSSL_RAND_PARAM_CIPHER                  OSSL_ALG_PARAM_CIPHER
+#define OSSL_RAND_PARAM_MAC                     OSSL_ALG_PARAM_MAC
+#define OSSL_RAND_PARAM_USE_DF                  "derivation_function"
+#define OSSL_RAND_PARAM_TEST_ENTROPY            "test_entropy"
+#define OSSL_RAND_PARAM_TEST_NONCE              "test_nonce"
```

Comment:
Which of these macros are generic for *all* possible RAND implementations (think beyond DRBG), and which of them are specific for DRBG-style implementations?  The macro names could vary accordingly.

Reply:
> Which of these macros are generic for all possible RAND implementations

You'll never know because you can't possibly know all possible RAND implementations.

I think OpenSSL should pick only DRBG and leave it at that.  Anyone else with something different (e.g., a noisy diode hardware implementation) can fake it to look like drbg.

Reply:
> You'll never know because you can't possibly know all possible RAND implementations.

So what you're saying is that exactly all those parameters are likely to be specific to our implementations, or plausibly for DRBG implementations.

> I think OpenSSL should pick only DRBG and leave it at that.

We've had discussions around that, and found it problematic to assume that too much for the libcrypto\<-\>provider interface.

Reply:
Based on my experiences replacing the rand with the old 1.0.2 DRBG code, yes, I believe all those parameters are required for DRBG.  I also think it is okay for openssl to say "we support this" and leave other things to others until they bring  a PR.

I am sure this will offend some of the project members, but OpenSSL has a history of providing the wrong kind of flexibility that just becomes a burden later on.  Look how much trouble ```RAND_METHOD``` is currently causing. Look at Billy's EC_POINT work. Look at the malloc layer, the "swappable error layer," and so on.

Reply:
Separating these into RAND and DRBG makes sense.  Most seem like they would be DRBG specific, the rest are more generally applicable to RNGs.  As @richsalz notes: our DRBGs require all of these (bar the two test items), but e.g. status is something all RNGs will have.